### PR TITLE
[ecs] Remove Printf usage

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Remove `fmt.Printf` usage in `ecs` util.

### Motivation

When the code path is executed, it would break the output of the
`hostname` command as expected by the process and trace agents.

### Additional notes

Long-term, we're planning to make other agents rely on core agent's internal API to avoid such issues (and also have more guarantees on hostname consistency across agents).

### CODE REVIEW

File: pkg/util/ecs/metadata/detection.go [hunk 1]
```diff
@@ -106,7 +106,6 @@ func testURLs(urls []string, timeout time.Duration) string {
 		}
 		var resp v1.Commands
 		if err := json.NewDecoder(r.Body).Decode(&resp); err != nil {
-			fmt.Printf("decode err: %s\n", err)
```
Comment (ahmed-mez) [inline pkg/util/ecs/metadata/detection.go:109]: shouldn't we replace it with a log message?

Comment (olivielpeau) [reply pkg/util/ecs/metadata/detection.go:109]: we don't log anything in the entire `detection` logic, so I'm assuming we don't need to here either (and I don't know if it'd become spammy if we did).

I can change this to a debug log though

Comment (ahmed-mez) [reply pkg/util/ecs/metadata/detection.go:109]: Adding a debug log could be useful yes, recently the ecs api broke the backward compatibility of one of its endpoints by returning a json with the same content but with different format, so we'll never know 🤷‍♂️ (cc @xornivore )

**Review by ahmed-mez — APPROVED** [review]

File: pkg/util/ecs/metadata/detection.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/util/ecs/metadata/detection.go:109]: 👍 addressed

Comment (ahmed-mez) [reply pkg/util/ecs/metadata/detection.go:109]: thank you!

**Pull request merged**